core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1211,16 +1285,10 @@ write_files:
                         break
                       fi
 
-                      if [ $((secs % SG_POLL_INTERVAL)) -eq 0 ]; then
-                        # Reset counter to avoid overflow
-                        secs=0
-
-                        # AutoScalingGroup lifecycle check
-                        state=$(asg describe-auto-scaling-instances --region "${REGION}" --instance-ids "${INSTANCE_ID}" | jq -r '.AutoScalingInstances[].LifecycleState')
-                        if [ "${state}" = Terminating:Wait ]; then
-                          termination_source=asg
-                          break
-                        fi
+                      # Termination annotation check
+                      termination_source=$(kubectl get nodes "${NODE_NAME}" -o json | jq -r '.metadata.annotations["kube-aws.coreos.com/node-termination-source"]')
+                      if [ "${termination_source}" == asg ]; then
+                        break
                       fi
                     done
 
```

Comment:
can we use `while sleep 10; do` or something like that so that it doesn't try to drain again immediately after `asg complete-lifecycle-action ` ?

It should still be a loop in case aws command failed or something else goes wrong and we need to repeat draining cycle.